internal/elasticsearch/security/role_mapping.go

```diff
@@ -0,0 +1,158 @@
+package security
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceRoleMapping() *schema.Resource {
+	roleMappingSchema := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The distinct name that identifies the role mapping, used solely as an identifier.",
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: "Mappings that have `enabled` set to `false` are ignored when role mapping is performed.",
+		},
+		"rules": {
+			Type:             schema.TypeString,
+			Required:         true,
+			DiffSuppressFunc: utils.DiffJsonSuppress,
+			Description:      "A list of mustache templates that will be evaluated to determine the role names that should granted to the users that match the role mapping rules. This matches fields of users, rules can be grouped into `all` and `any` top level keys.",
```

Comment:
This description is for role_templates which is missing from the resource schema.